packages/adapters/wagmi/src/client.ts

```diff
@@ -213,21 +213,20 @@ export class WagmiAdapter extends AdapterBlueprint {
 
   private setupWatchers() {
     watchAccount(this.wagmiConfig, {
-      onChange: accountData => {
+      onChange: (accountData, prevAccountData) => {
         if (accountData.status === 'disconnected') {
           this.emit('disconnect')
         }
         if (accountData.status === 'connected') {
-          if (accountData.address) {
+          if (accountData.address !== prevAccountData?.address) {
             this.setupWatchPendingTransactions()
 
             this.emit('accountChanged', {
-              address: accountData.address,
-              chainId: accountData.chainId
+              address: accountData.address
             })
           }
 
-          if (accountData.chainId) {
+          if (accountData.chainId !== prevAccountData?.chainId) {
```

Comment:
What about the other clients? Is it going to have same functionality for the other clients?

Reply:
Given wagmi triggers all of these updates at the same time, this is to be able to individually re-fire the events. On others adapters they are handled individually. SAdly there's no way to get the "previous" data else I would apply the same optimization there :\